assets/src/edit-story/components/form/numeric.js

```diff
@@ -99,38 +99,84 @@ function Numeric({
 }) {
   const isMultiple = value === MULTIPLE_VALUE;
   const placeholder = isMultiple ? __('multiple', 'web-stories') : '';
-  const [dot, setDot] = useState(false);
   const [inputValue, setInputValue] = useState(value);
   const ref = useRef();
+  const targetFormRef = useRef(null);
 
-  useEffect(() => setInputValue(value), [value]);
+  const isIterimValue = useCallback(
+    (val) =>
+      // Allow a single "-" if negative values are allowed
+      (canBeNegative && val === '-') ||
+      // Allow string of 0s
+      /^[0]+$/.test(val) ||
+      // Allow starting and ending with decimal point if float
+      (float && (val.startsWith(DECIMAL_POINT) || val.endsWith(DECIMAL_POINT))),
+    [float, canBeNegative]
+  );
+
+  const getProcessedInput = useCallback(
+    (val) => {
+      const indexOfDot = val.indexOf(DECIMAL_POINT);
+      const fixedPoint =
+        float && indexOfDot > -1 ? val.length - (indexOfDot + 1) : 0;
+
+      const valueAsNumber = float
+        ? parseFloat(val).toFixed(fixedPoint)
+        : parseInt(val);
+
+      const signedNumber = !canBeNegative
+        ? Math.abs(valueAsNumber)
+        : valueAsNumber;
+
+      return {
+        valueAsNumber,
+        signedNumber,
+        fixedPoint,
+      };
```

Comment:
If possible this feels like a function we could hoist and just pass in `float` and `canBeNegative` that way it's easier to test this in isolation. Also this is a lot of logic to memoize.

Reply:
I can dig it, will hoooist it up!